File: _code-samples/send-xrp/js/send-xrp.js
// Dependencies for Node.js.
// In browsers, use a <script> tag instead.
if (typeof module !== "undefined") {
  // Use var here because const/let are block-scoped to the if statement.
  var xrpl = require('xrpl')
}

// Example credentials
const wallet = xrpl.Wallet.fromSeed("sn3nxiW7v8KXzPzAqzyHXbSSKNuN9")
console.log(wallet.address) // rMCcNuTcajgw7YTgBy1sys3b89QqjUrMpH

// Connect -------------------------------------------------------------------
async function main() {
  console.log("Connecting to Testnet...")
  const client = new xrpl.Client('wss://s.altnet.rippletest.net:51233')
  await client.connect()

  // Get credentials from the Testnet Faucet -----------------------------------
  console.log("Getting a wallet from the Testnet faucet...")
  const {wallet, balance} = await client.fundWallet()

  // Prepare transaction -------------------------------------------------------
  const prepared = await client.autofill({
    "TransactionType": "Payment",
    "Account": wallet.address,
    "Amount": xrpl.xrpToDrops("22"),
    "Destination": "rPT1Sjq2YGrBMTttX4GZHjKu9dyfzbpAYe"
  })
  const max_ledger = prepared.LastLedgerSequence
  console.log("Prepared transaction instructions:", prepared)
  console.log("Transaction cost:", xrpl.dropsToXrp(prepared.Fee), "XRP")
  console.log("Transaction expires after ledger:", max_ledger)

  // Sign prepared instructions ------------------------------------------------
  const signed = wallet.sign(prepared)
  console.log("Identifying hash:", signed.hash)
  console.log("Signed blob:", signed.tx_blob)

  // Submit signed blob --------------------------------------------------------
  const tx = await client.submitAndWait(signed.tx_blob)

  // Wait for validation -------------------------------------------------------
  // submitAndWait() handles this automatically, but it can take 4-7s.

  // Check transaction results -------------------------------------------------
  console.log("Transaction result:", tx.result.meta.TransactionResult)
  console.log("Balance changes:", JSON.stringify(xrpl.getBalanceChanges(tx.result.meta), null, 2))

  // End of main()
  client.disconnect()
}

main()

File: _code-samples/send-xrp/ruby/send-xrp.rb
# Example credentials ----------------------------------------------------------
require 'xrpl-ruby'
require 'json'
test_wallet = Wallet::Wallet.from_seed('sn3nxiW7v8KXzPzAqzyHXbSSKNuN9')
puts test_wallet.classic_address # "rMCcNuTcajgw7YTgBy1sys3b89QqjUrMpH"

# Connect ----------------------------------------------------------------------
client = XRPL::Client.new(:testnet)
client.connect!

# Get credentials from the Testnet faucet --------------------------------------
# For production, instead create a Wallet instance as above.
test_wallet = XRPL.fund_wallet(client)[:wallet]
puts "Funded account: #{test_wallet.classic_address}"

# Prepare transaction ----------------------------------------------------------
# Amounts are expressed in drops of XRP, where 1 XRP = 1,000,000 drops.
payment = {
  'TransactionType' => 'Payment',
  'Account' => test_wallet.classic_address,
  'Amount' => '22000000', # 22 XRP
  'Destination' => 'rPT1Sjq2YGrBMTttX4GZHjKu9dyfzbpAYe'
}
prepared = client.autofill(payment)
puts "Prepared transaction: #{prepared}"

# Sign transaction -------------------------------------------------------------
signed = test_wallet.sign(prepared)
tx_id = signed['hash']
max_ledger = prepared['LastLedgerSequence']
puts "Identifying hash: #{tx_id}"
puts "Transaction expires after ledger: #{max_ledger}"

# Submit transaction -----------------------------------------------------------
# submit_and_wait deterministically re-signs the already-autofilled transaction
# (producing the same hash) and waits until it is in a validated ledger.
tx_response = client.submit_and_wait(prepared, wallet: test_wallet, autofill: false)

# Wait for validation ----------------------------------------------------------
# submit_and_wait handles this automatically, but it can take 4-7 seconds.

# Check transaction results ----------------------------------------------------
puts JSON.pretty_generate(tx_response)
puts "Explorer link: https://testnet.xrpl.org/transactions/#{tx_id}"
metadata = tx_response.dig('result', 'meta') || {}
puts "Result code: #{metadata['TransactionResult']}" if metadata['TransactionResult']
